### PR TITLE
chore(typo): improve typescript declaration

### LINKED ISSUE #41882: npm安装后node_modules/antd/lib/menu/OverrideContext.d.ts:11:16 导出的OverrideContext未定义
### Reproduction link

[https://u.ant.design/codesandbox-repro](https://u.ant.design/codesandbox-repro)

### Steps to reproduce

npm安装antd@5.4.2
打开node_modules/antd/lib/menu/OverrideContext.d.ts:11:16 

export default OverrideContext; //OverrideContext未定义

### What is expected?

node_modules/antd/lib/menu/OverrideContext.d.ts:11:16 

export default OverrideContextProps; //应该是导出OverrideContextProps才对

### What is actually happening?

参看重现步骤

| Environment | Info |
| --- | --- |
| antd | 5.4.2 |
| React | 17.0.2 |
| System | mac os 12.6 |
| Browser | chrome |

### PR DESCRIPTION
**Pull request by Wxh16144**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [x] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

close #41882 

### 💡 Background and solution

 **解决方案:** 将私有的方法添加 `@internal` 标识, 已跳过 TS 生成描述文件. 建议之后都统一使用 `@internal` 而非 `@private`

[构建后产物对比](https://github.com/Wxh16144/check-antd-typescript-declaration/compare/85e6f3a98ab8fda545f84342377c21fa285a33b0...24be3d35fba770a3135bd7e5d0c54c32725f0616)

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     Improved JSDoc handling of internal methods in TS description files      |
| 🇨🇳 Chinese |      改进 TS 描述文件对内部方法的 JSDoc 处理     |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at d340911</samp>

Marked some internal properties and objects as `@internal` in the JSDoc comments of `components/theme` files. This improves the documentation and the TypeScript declaration consistency and accuracy.

### <samp>🤖 Generated by Copilot at d340911</samp>

*  Mark `defaultConfig`, `Wave`, `statistic`, and `_statistic_build_` as `@internal` in JSDoc comments to match TypeScript declaration file ([link](https://github.com/ant-design/ant-design/pull/41907/files?diff=unified&w=0#diff-a1691067c7ae7cc4ec252c57ae55fccd18ee928b026091db042495c6cd43e049L21-R21), [link](https://github.com/ant-design/ant-design/pull/41907/files?diff=unified&w=0#diff-1d8c7d2026da8b02ccc59970cea90849086b133e4b8f55d59ae475ac814d9e1bL117-R117), [link](https://github.com/ant-design/ant-design/pull/41907/files?diff=unified&w=0#diff-0dc90a406b302dfe27fbf4738fdb36cdef989d720ee19295495254f2d8a2115dL37-R37), [link](https://github.com/ant-design/ant-design/pull/41907/files?diff=unified&w=0#diff-0dc90a406b302dfe27fbf4738fdb36cdef989d720ee19295495254f2d8a2115dL43-R43))

### CODE REVIEW

File: components/config-provider/index.tsx [hunk 1]
```diff
@@ -336,7 +336,7 @@ const ProviderChildren: React.FC<ProviderChildrenProps> = (props) => {
 };
 
 const ConfigProvider: React.FC<ConfigProviderProps> & {
-  /** @private internal Usage. do not use in your production */
+  /** @internal internal Usage. do not use in your production */
```
Comment (zombieJ) [inline components/config-provider/index.tsx:339]: 只用fix 原本是 @internal 还有导出的即可，有的写 Internal usage 但是 ProComponent 是要用的。

**Comment by Wxh16144** [comment]
ready~

File: components/theme/interface/components.ts [hunk 2]
```diff
@@ -114,6 +114,6 @@ export interface ComponentTokenMap {
   QRCode?: QRCodeComponentToken;
   App?: AppComponentToken;
 
-  /** @private Internal TS definition. Do not use. */
+  /** @internal Internal TS definition. Do not use. */
```
Comment (zombieJ) [inline components/theme/interface/components.ts:117]: 这个不是 internal，快乐工作主题是个三方包，这个藏了就用不了了

File: components/theme/index.ts [hunk 3]
```diff
@@ -18,7 +18,7 @@ function useToken() {
 export { type GlobalToken };
 
 export default {
-  /** @private Test Usage. Do not use in production. */
+  /** @internal Test Usage. Do not use in production. */
```
Comment (zombieJ) [inline components/theme/index.ts:21]: 这个会导致三方包不能消费，这个不太能 internal 掉

Comment (Wxh16144) [inline components/theme/index.ts:21]:
```suggestion
  /** @private Test Usage. Do not use in production. */
```

File: components/theme/interface/components.ts (hunk 2 above)
Comment (Wxh16144) [inline components/theme/interface/components.ts:117]:
```suggestion
  /** @pravite Internal TS definition. Do not use. */
```

**Comment by afc163** [comment]
any progress?

**Comment by Wxh16144** [comment]
> any progress?

readey, need review.

**Review by zombieJ — APPROVED** [review]

**Pull request merged**